src/main/generic/consensus/blockchain/Blockchain.js

```diff
@@ -435,48 +435,42 @@ class Blockchain extends Observable {
     }
 
     async getNextCompactTarget(chain) {
+        // The difficulty is adjusted every blocks.
         chain = chain || this._mainChain;
 
-        // The difficulty is adjusted every DIFFICULTY_ADJUSTMENT_BLOCKS blocks.
-        if (chain.height % Policy.DIFFICULTY_ADJUSTMENT_BLOCKS === 0) {
-            // If the given chain is the main chain, get the last DIFFICULTY_ADJUSTMENT_BLOCKS
-            // blocks via this._mainChain, otherwise fetch the path.
-            let startHash;
-            if (chain === this._mainChain) {
-                const startHeight = Math.max(this._mainPath.length - Policy.DIFFICULTY_ADJUSTMENT_BLOCKS, 0);
-                startHash = this._mainPath[startHeight];
-            } else {
-                const path = await this._fetchPath(chain.head, Policy.DIFFICULTY_ADJUSTMENT_BLOCKS - 1);
-                startHash = path[0];
-            }
-
-            // Compute the actual time it took to mine the last DIFFICULTY_ADJUSTMENT_BLOCKS blocks.
-            const startChain = await this._store.get(startHash.toBase64());
-            const actualTime = chain.head.timestamp - startChain.head.timestamp;
+        // If the given chain is the main chain, get the last DIFFICULTY_BLOCK_WINDOW
+        // blocks via this._mainChain, otherwise fetch the path.
+        let startHash;
+        if (chain === this._mainChain) {
+            const startHeight = Math.max(this._mainPath.length - Policy.DIFFICULTY_BLOCK_WINDOW, 0);
+            startHash = this._mainPath[startHeight];
+        } else {
+            const path = await this._fetchPath(chain.head, Policy.DIFFICULTY_BLOCK_WINDOW - 1);
+            startHash = path[0];
+        }
 
-            // Compute the target adjustment factor.
-            const expectedTime = Policy.DIFFICULTY_ADJUSTMENT_BLOCKS * Policy.BLOCK_TIME;
-            let adjustment = actualTime / expectedTime;
+        // Compute the actual time it took to mine the last DIFFICULTY_BLOCK_WINDOW blocks.
+        const startChain = await this._store.get(startHash.toBase64()); // chain head is Policy.DIFFICULTY_BLOCK_WINDOW back
+        const actualTime = chain.head.timestamp - startChain.head.timestamp;
 
-            // Clamp the adjustment factor to [0.25, 4].
-            adjustment = Math.max(adjustment, 0.25);
-            adjustment = Math.min(adjustment, 4);
+        // Compute the target adjustment factor.
+        const expectedTime = Policy.DIFFICULTY_BLOCK_WINDOW * Policy.BLOCK_TIME;
+        let adjustment = actualTime / expectedTime;
 
-            // Compute the next target.
-            const currentTarget = chain.head.target;
-            let nextTarget = currentTarget * adjustment;
+        // Clamp the adjustment factor to [0.5, 2].
+        adjustment = Math.max(adjustment, 0.5);
```

Comment:
Perhaps we should extract those magic numbers and represent them in our Policy.
Also, since we adjust difficulty every block, 0.5 and 2 might be too large.